### PR TITLE
project structure and table inserts

### PR DESCRIPTION
**Pull request by BenGriffith**

### CODE REVIEW

File: src/cloud_run/utils/table_logger.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            file_handler = logging.FileHandler(log_file)
+            file_handler.setLevel(level)
+
+            formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+            file_handler.setFormatter(formatter)
+
+            table_logger.addHandler(file_handler)
+
+        return table_logger
+    else:
```
Comment (bbelderbos) [inline src/cloud_run/utils/table_logger.py:19]: could do an early return, but that might be just my preference

File: src/cloud_run/utils/helpers.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    make: str
+    model: str
+    category: str
+    license_plate: str
+    vin: str
+    state: str
+    primary_color: str
+    tollway_state: str
+    tollway_name: str
+    timestamp: str
```
Comment (bbelderbos) [inline src/cloud_run/utils/helpers.py:16]: datetime?

File: src/cloud_run/utils/constants.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import os
+
+PROJECT_ID = "playground-413722"  # os.getenv("PROJECT_ID")
+SUBSCRIPTION_ID = os.getenv("SUBSCRIPTION_ID")
+DATASET_ID = "tollway_traffic"  # os.getenv("DATASET_ID")
```
Comment (bbelderbos) [inline src/cloud_run/utils/constants.py:5]: or `os.getenv("DATASET_ID", "tollway_traffic")`?

File: src/cloud_run/tables/fact_tollway_event.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    tollway_logger = setup_logger(True)
+
+    try:
+        fact_tollway_event_errors = bigquery_client.insert_rows_json(table_ref, [fact_tollway_event_row])
+        tollway_logger.info(f"Successfully inserted event {event_id} into BigQuery")
+    except GoogleAPIError as ge:
+        tollway_logger.error(f"Encountered errors while inserting event {event_id} into BigQuery: {ge}")
+        raise
+    except Exception as e:
+        tollway_logger.error(
```
Comment (bbelderbos) [inline src/cloud_run/tables/fact_tollway_event.py:30]: can also do logger.exception

Comment (bbelderbos) [reply src/cloud_run/tables/fact_tollway_event.py:30]: seems handling is the same so could just use Exception or `except (A, B) as e`

File: src/cloud_run/tables/dim_tollway.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    dim_tollway_row = {
+        "tollway_id": hash_string(message_data.tollway_name),
+        "tollway_name": message_data.tollway_name,
+    }
+
+    table_ref = bigquery_client.dataset(DATASET_ID).table(TABLES["dim_tollway"])
+    dim_tollway_errors = bigquery_client.insert_rows_json(table_ref, [dim_tollway_row])
+
+    if dim_tollway_errors:
+        ...
```
Comment (bbelderbos) [inline src/cloud_run/tables/dim_tollway.py:34]: why the ellipsis?

File: src/cloud_run/tables/fact_tollway_event.py (hunk 4 above)
Comment (bbelderbos) [reply src/cloud_run/tables/fact_tollway_event.py:30]: might not apply anymore as you made changes to this in #15  ...

File: src/cloud_run/utils/table_logger.py (hunk 1 above)
Comment (BenGriffith) [reply src/cloud_run/utils/table_logger.py:19]: good suggestion. I'm starting to like use of early return more.

File: src/cloud_run/utils/helpers.py (hunk 2 above)
Comment (BenGriffith) [reply src/cloud_run/utils/helpers.py:16]: possibly. I think other program that generates the messages has str type for the timestamp key. I'll doublecheck to make sure using correct type.

File: src/cloud_run/utils/constants.py (hunk 3 above)
Comment (BenGriffith) [reply src/cloud_run/utils/constants.py:5]: this commit/change shouldn't have made it in. I have the values as literals right now to help with testing. Once I'm further along, if possible, thinking to define environment variables in Terraform script.

File: src/cloud_run/tables/fact_tollway_event.py (hunk 4 above)
Comment (BenGriffith) [reply src/cloud_run/tables/fact_tollway_event.py:30]: will look into this more. Thanks.

File: src/cloud_run/tables/dim_tollway.py (hunk 5 above)
Comment (BenGriffith) [reply src/cloud_run/tables/dim_tollway.py:34]: know it's not best practice but using the ellipsis to simply provide some structure for future code development/reminder so I don't forget about it

**Pull request merged**